modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1249 @@
+
+      on this {
+        _enter();
+
+        var count = 0;
+
+        for item in this do
+          if x == item then
+            count += 1;
+
+        result = count;
+
+        _leave();
+      }
+
+      return result;
+    }
+
+    pragma "no doc"
+    proc ref sort(comparator: ?rec=Sort.defaultComparator) {
+      compilerError("unrolledLinkedList doesn't support sort");
+      return;
+    }
+
+    /*
+      Index this unrolledLinkedList via subscript. Returns a reference to the element at a
+      given index in this unrolledLinkedList.
+
+      :arg i: The index of the element to access.
+
+      .. warning::
+
+        Use of the `this` method with an out of bounds index (while bounds
+        checking is on) will cause the currently running program to halt.
+
+      :return: An element from this unrolledLinkedList.
+    */
+    proc ref this(i: int) ref {
+      if boundsChecking && !_withinBounds(i) {
+        const msg = "Invalid unrolledLinkedList index: " + i:string;
+        boundsCheckHalt(msg);
+      }
+
+      ref result = _getRef(i);
+      return result;
+    }
+
+    pragma "no doc"
+    proc const ref this(i: int) const ref {
+      if boundsChecking && !_withinBounds(i) {
+        const msg = "Invalid unrolledLinkedList index: " + i:string;
+        halt(msg);
+      }
+
+      const ref result = _getRef(i);
+      return result;
+    }
+
+    /*
+      Iterate over the elements of this unrolledLinkedList.
+
+      :yields: A reference to one of the elements contained in this unrolledLinkedList.
+    */
+    iter these() ref {
+      var cur = _head;
+      while cur != nil {
+        for i in 0..#cur!.size {
+          yield cur!.data[i];
+        }
+        cur = cur!.next;
+      }
+    }
+
+    /*
+      Write the contents of this unrolledLinkedList to a channel.
+
+      :arg ch: A channel to write to.
+    */
+    proc writeThis(ch: channel) throws {
+      _enter();
+      
+      ch <~> "[";
+
+      var first = true;
+
+      for x in this {
+        if !first then ch <~> ", ";
+        else first = false;
+        ch <~> x;
+      }
+
+      ch <~> "]";
+
+      _leave();
+    }
+
+    /*
+      Returns `true` if this unrolledLinkedList contains zero elements.
+
+      :return: `true` if this unrolledLinkedList is empty.
+      :rtype: `bool`
+    */
+    proc const isEmpty(): bool {
+      return size == 0;
+    }
+
+    /*
+      The current number of elements contained in this unrolledLinkedList.
+    */
+    inline proc const size {
+      var result = 0;
+
+      on this {
+        _enter();
+        result = _size;
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Returns the unrolledLinkedList's legal indices as the range ``0..<this.size``.
+
+      :return: ``0..<this.size``
+      :rtype: `range`
+    */
+    proc indices {
+      return 0..<this.size;
+    }
+
+    /*
+      Returns a new DefaultRectangular array containing a copy-initialized var of 
```

Comment:
var? version?

Reply:
Hmm...I suppose it's variable. This line must be copied from somewhere but I don't remember where. I remember @cassella once suggested changing the wording of this line. Any ideas?

Reply:
I don't remember making any previous suggestions here, but it's been a while.  Could it just be "... array containing a copy of each of the elements..."?